src/profiler/storage_profiler.h

```diff
@@ -52,6 +54,12 @@ class DeviceStorageProfiler {
       if (prof->IsProfiling(profiler::Profiler::kMemory)) {
         Init();
         const size_t idx = prof->DeviceIndex(handle.ctx.dev_type, handle.ctx.dev_id);
+        // sleep for a few seconds until the mem_counters_ is fully initialized
+        size_t timeout = 1000;
```

Comment:
Should we throw an error if the timeout is not sufficient instead of using uninitialized data?

Reply:
Yes, line 63 does that: `CHECK_LT(idx, mem_counters_.size()) << "Invalid device index: " << idx;`